sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -750,10 +750,13 @@ private <T> Slice<T> sliceQuery(SqlQuerySpec querySpec,
 
         Flux<FeedResponse<JsonNode>> feedResponseFlux;
         if (pageable instanceof CosmosPageRequest) {
+            int contentSize = pageable.getPageSize();
```

Comment:
Can we please add some comments here to explain this logic of adding the offset to content size.
And why are we passing the content size to byPage() API?

Reply:
Done.

Reply:
Does this handle going straight from page 1 to 3 directly? Our experience with the continuation token prevented that scenario historically.

Reply:
Let me add a new unit test for that scenario!

Reply:
Azure Cosmos DB currently only supports pagination starting at the first page, and due to the use of continuation tokens you cannot skip pages in this way (going from page 1 to 3 directly).
https://docs.microsoft.com/en-us/azure/cosmos-db/sql/sql-query-pagination

Reply:
This PR does not meet the needs of the issues then and just continues Azure's support of "continuation" and not "pagination".

Reply:
@azizabah that's correct. We would like to keep the PRs short for easier review and reduce the chances of any regressions. 
We will re-prioritize the page number implementation in another PR.